project/LinkerdBuild.scala

```diff
@@ -233,6 +233,56 @@ object LinkerdBuild extends Base {
 
   val ConfigFileRE = """^(.*)\.yaml$""".r
 
+  val baseExecScript =
+    """|#!/bin/sh
+       |
+       |jars="$0"
+       |HSPREF_SETTING=$([ ! -z "$ENABLE_HSPREF" ] && echo "" || echo "-XX:+PerfDisableSharedMem")
+       |if [ -n "$NAMERD_HOME" ] && [ -d $NAMERD_HOME/plugins ]; then
```

Comment:
Considering that this script is specific to Namerd, we may have issues running Linkerd i.e. Linkerd plugins since it does not reference `$L5D_HOME`. Consider having just the gc check session in the `baseExecScript` variable. We probably would also need to change the name of the val as well.
```diff
diff --git a/project/LinkerdBuild.scala b/project/LinkerdBuild.scala
index aa6726b51..8e69cb010 100644
--- a/project/LinkerdBuild.scala
+++ b/project/LinkerdBuild.scala
@@ -232,6 +232,35 @@ object LinkerdBuild extends Base {
   }
 
   val ConfigFileRE = """^(.*)\.yaml$""".r
+  
+  val baseGcLogCheck = """
+    |# Check Java version for use in GC_LOG_OPTION and DEFAULT_JVM_OPTIONS
+    |LOCAL_JAVA_VERSION=$("${JAVA_HOME:-/usr}"/bin/java -version 2>&1 | sed 's/.*version "\([0-9]*\)\..*/\1/; 1q')
+    |
+    |mkdir -p "$GC_LOG" && [ -w "$GC_LOG" ]
+    |
+    |if [ $? -ne 0 ]; then
+    |  echo "GC_LOG must be set to a directory that user [$USER] has write permissions on.\
+    | Unable to use [$GC_LOG] for GC logging."
+    |else
+    |  if [ "$LOCAL_JAVA_VERSION" -ge 9 ]; then
+    |    GC_LOG_OPTION="-Xlog:gc*,gc+age=trace,gc+heap=debug,gc+promotion=trace,safepoint:file=${GC_LOG}/gc.log::filecount=10,filesize=10000:time"
+    |  else
+    |    GC_LOG_OPTION="
+    |      -XX:+PrintGCDetails
+    |      -XX:+PrintGCDateStamps
+    |      -XX:+PrintHeapAtGC
+    |      -XX:+PrintTenuringDistribution
+    |      -XX:+PrintGCApplicationStoppedTime
+    |      -XX:+PrintPromotionFailure
+    |      -Xloggc:${GC_LOG}/gc.log
+    |      -XX:+UseGCLogFileRotation
+    |      -XX:NumberOfGCLogFiles=10
+    |      -XX:GCLogFileSize=10M"
+    |  fi
+    |fi
+    |
+  """.stripMargin
 
   val execScriptJvmOptions =
     """|DEFAULT_JVM_OPTIONS="$DEFAULT_JVM_OPTIONS                        \
@@ -373,34 +402,9 @@ object LinkerdBuild extends Base {
          |  GC_LOG="/var/log/namerd"
          |fi
          |
-         |# Check Java version for use in GC_LOG_OPTION and DEFAULT_JVM_OPTIONS
-         |LOCAL_JAVA_VERSION=$("${JAVA_HOME:-/usr}"/bin/java -version 2>&1 | sed 's/.*version "\([0-9]*\)\..*/\1/; 1q')
-         |
-         |mkdir -p "$GC_LOG" && [ -w "$GC_LOG" ]
-         |
-         |if [ $? -ne 0 ]; then
-         |  echo "GC_LOG must be set to a directory that user [$USER] has write permissions on.\
-         | Unable to use [$GC_LOG] for GC logging."
-         |else
-         |  if [ "$LOCAL_JAVA_VERSION" -ge 9 ]; then
-         |    GC_LOG_OPTION="-Xlog:gc*,gc+age=trace,gc+heap=debug,gc+promotion=trace,safepoint:file=${GC_LOG}/gc.log::filecount=10,filesize=10000:time"
-         |  else
-         |    GC_LOG_OPTION="
-         |      -XX:+PrintGCDetails
-         |      -XX:+PrintGCDateStamps
-         |      -XX:+PrintHeapAtGC
-         |      -XX:+PrintTenuringDistribution
-         |      -XX:+PrintGCApplicationStoppedTime
-         |      -XX:+PrintPromotionFailure
-         |      -Xloggc:${GC_LOG}/gc.log
-         |      -XX:+UseGCLogFileRotation
-         |      -XX:NumberOfGCLogFiles=10
-         |      -XX:GCLogFileSize=10M"
-         |  fi
-         |fi
-         |
-         |""" +
-      execScriptJvmOptions +
+         |""" + 
+        baseGcLogCheck +
+        execScriptJvmOptions +
       """|exec "${JAVA_HOME:-/usr}/bin/java" -XX:+PrintCommandLineFlags \
          |     ${JVM_OPTIONS:-$DEFAULT_JVM_OPTIONS} $HSPREF_SETTING -cp $jars -server \
          |     io.buoyant.namerd.Main "$@"
@@ -468,33 +472,8 @@ object LinkerdBuild extends Base {
          |  GC_LOG="/var/log/namerd"
          |fi
          |
-         |# Check Java version for use in GC_LOG_OPTION and DEFAULT_JVM_OPTIONS
-         |LOCAL_JAVA_VERSION=$("${JAVA_HOME:-/usr}"/bin/java -version 2>&1 | sed 's/.*version "\([0-9]*\)\..*/\1/; 1q')
-         |
-         |mkdir -p "$GC_LOG" && [ -w "$GC_LOG" ]
-         |
-         |if [ $? -ne 0 ]; then
-         |  echo "GC_LOG must be set to a directory that user [$USER] has write permissions on.\
-         | Unable to use [$GC_LOG] for GC logging."
-         |else
-         |  if [ "$LOCAL_JAVA_VERSION" -ge 9 ]; then
-         |    GC_LOG_OPTION="-Xlog:gc*,gc+age=trace,gc+heap=debug,gc+promotion=trace,safepoint:file=${GC_LOG}/gc.log::filecount=10,filesize=10000:time"
-         |  else
-         |    GC_LOG_OPTION="
-         |      -XX:+PrintGCDetails
-         |      -XX:+PrintGCDateStamps
-         |      -XX:+PrintHeapAtGC
-         |      -XX:+PrintTenuringDistribution
-         |      -XX:+PrintGCApplicationStoppedTime
-         |      -XX:+PrintPromotionFailure
-         |      -Xloggc:${GC_LOG}/gc.log
-         |      -XX:+UseGCLogFileRotation
-         |      -XX:NumberOfGCLogFiles=10
-         |      -XX:GCLogFileSize=10M"
-         |  fi
-         |fi
-         |
          |""" +
+      baseGcLogCheck +
       execScriptJvmOptions +
       """|if read -t 0; then
          |  CONFIG_INPUT=`cat`
@@ -713,33 +692,8 @@ object LinkerdBuild extends Base {
          |  GC_LOG="/var/log/linkerd"
          |fi
          |
-         |# Check Java version for use in GC_LOG_OPTION and DEFAULT_JVM_OPTIONS
-         |LOCAL_JAVA_VERSION=$("${JAVA_HOME:-/usr}"/bin/java -version 2>&1 | sed 's/.*version "\([0-9]*\)\..*/\1/; 1q')
-         |
-         |mkdir -p "$GC_LOG" && [ -w "$GC_LOG" ]
-         |
-         |if [ $? -ne 0 ]; then
-         |  echo "GC_LOG must be set to a directory that user [$USER] has write permissions on.\
-         | Unable to use [$GC_LOG] for GC logging."
-         |else
-         |  if [ "$LOCAL_JAVA_VERSION" -ge 9 ]; then
-         |    GC_LOG_OPTION="-Xlog:gc*,gc+age=trace,gc+heap=debug,gc+promotion=trace,safepoint:file=${GC_LOG}/gc.log::filecount=10,filesize=10000:time"
-         |  else
-         |    GC_LOG_OPTION="
-         |      -XX:+PrintGCDetails
-         |      -XX:+PrintGCDateStamps
-         |      -XX:+PrintHeapAtGC
-         |      -XX:+PrintTenuringDistribution
-         |      -XX:+PrintGCApplicationStoppedTime
-         |      -XX:+PrintPromotionFailure
-         |      -Xloggc:${GC_LOG}/gc.log
-         |      -XX:+UseGCLogFileRotation
-         |      -XX:NumberOfGCLogFiles=10
-         |      -XX:GCLogFileSize=10M"
-         |  fi
-         |fi
-         |
          |""" +
+      baseGcLogCheck +
       execScriptJvmOptions +
       """|exec "${JAVA_HOME:-/usr}/bin/java" -XX:+PrintCommandLineFlags \
          |     ${JVM_OPTIONS:-$DEFAULT_JVM_OPTIONS} $HSPREF_SETTING -cp $jars -server \

```

(I think I may have left out somethings from your change in this diff but hopefully this conveys what I was thinking)